src/ConfigFactory.Core/ConfigModule.cs

```diff
@@ -50,8 +50,8 @@ public ConfigModule()
         Properties = ConfigProperties.Generate<T>();
     }
 
-    IConfigModule IConfigModule.Load() => Load();
-    private static T Load()
+    public virtual IConfigModule Load() => DefaultLoad();
+    private static T DefaultLoad()
```

Comment:
The DefaultLoad is the original static `Load()` method from your code - it is static since it is used in the static "Shared" property, but it can serve as a default implementation of the Loading logic in the (newly virtualized) Load() method.
It is a good serialization/deserialization way that might serve great for the basic cases.

(But virutalization of the Load() method itself now brings the desired possibility to override this behavior with some own more complicated serialization / deserialization logic)

Reply:
I think it's a bit risky to have the `Shared` property using a default implementation since most people will likely expect Shared to use the `Load()` method.

I'm not sure what the best solution to that is though, so I think it's fine for now, I might come back to it later.